src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp

```diff
@@ -0,0 +1,409 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Optimizer/PassTemporaryToStencilFunction.h"
+#include "dawn/Optimizer/AccessComputation.h"
+#include "dawn/Optimizer/OptimizerContext.h"
+#include "dawn/Optimizer/StatementMapper.h"
+#include "dawn/Optimizer/Stencil.h"
+#include "dawn/Optimizer/StencilInstantiation.h"
+#include "dawn/SIR/AST.h"
+#include "dawn/SIR/ASTVisitor.h"
+#include "dawn/SIR/SIR.h"
+
+namespace dawn {
+
+/// @brief this pass will identify temporaries of a stencil and replace their pre-computations
+/// by a stencil function. Each reference to the temporary is later replaced by the stencil function
+/// call.
+/// * Input: well formed SIR and IIR with the list of mss/stages, temporaries used and
+/// <statement,accesses> pairs
+/// * Output: modified SIR, new stencil functions are inserted and calls. Temporary fields are
+/// removed. New stencil functions instantiations are inserted into the IIR. <statement,accesses>
+/// pairs are recomputed
+
+namespace {
```

Comment:
yes, I would actually avoid both and use rather static methods of a helper class. The helper class would provide a context of where the action is being applied. In this case, it could actually be a static method of `Interval::toSIRInterval`. 
I think I made it a C function cause I didnt understand why we have two intervals, Interval and sir::Interval that are the same, and thought we could just retain one. Do you know any reason why we would like to keep both of them?